app/components/select-percentage.tsx

```diff
@@ -0,0 +1,183 @@
+"use client";
+import React, { useState } from "react";
+import { X } from "lucide-react";
+
+/**
+    Select percentage
+*/
+type PercentageValue = 25 | 50 | 75 | 100;
+const SelectPercentage = () => {
+  const [baseToken, setBaseToken] = useState("");
+  const [amount, setAmount] = useState("");
+  const [percent, setPercent] = useState(0);
+
+  const numberRegex = /^[0-9]*[.,]?[0-9]*$/;
+
+  const handleAmountChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const value = e.target.value;
+    if (numberRegex.test(value)) {
+      setAmount(value);
+      console.log(amount);
+    }
+  };
+
+  const handleBaseTokenChange = (value: string) => {
+    if (value == baseToken) {
+      setBaseToken("");
+    } else {
+      setBaseToken(value);
+    }
+  };
+
+  return (
+    <div className="py-16 h-auto flex flex-col items-center justify-center mx-auto w-full px-4">
+      <div className="w-full max-w-[800px] px-4 sm:px-8 py-12 border-[2px] border-[#170F2E] flex flex-col items-center justify-center space-y-4 rounded-xl">
+        <div className="flex flex-col items-center justify-center relative w-full sm:max-w-[90%]">
+          <div className="text-center w-fit flex-none font-[600] text-[20px] sm:text-[24px] px-4">
+            Select token(s) to auto-swap from
+          </div>
+          <div className="absolute right-0 w-4 h-full flex-none flex items-center justify-center mx-auto">
+            <X className="cursor-pointer" />
+          </div>
+        </div>
+        <div className="w-full py-6 px-4 max-w-[620px] text-[14px] sm:text-[16px] leading-[22px] text-[#A199B8] text-center">
+          You can select the percentage you want auto-swapped to your preferred
+          base token or input the percentage yourself.
+        </div>
+        <div className="w-full sm:max-w-[90%] flex flex-col items-start justify-start space-y-6">
+          <div className="w-full rounded-full flex flex-row items-center justify-between space-x-2 bg-[#100827]">
+            <input
```

Comment:
When ever a user types into this input field, the selected percentage should unselect. And when a percentage is selected, the input field value should go to "".